<?xml version="1.0"?>
<launch>

    <!-- send urdf to param server -->
    <param name="robot_description" textfile="$(find custom_arm)/urdf/custom_arm.urdf"/>

    <!-- Non-actuated joints -->
    <node name="RSP" pkg="robot_state_publisher" type="robot_state_publisher" output="screen"/>
    
    <!-- Launch Gazebo World -->
    <include file="$(find gazebo_ros)/launch/empty_world.launch"/>
  
    <!-- Spawn robot -->
    <node name="spawn_urdf" pkg="gazebo_ros" type="spawn_model" args="-param robot_description -urdf -model custom_arm"/>
    
    <!-- Load controller config -->
    <rosparam command="load" file="$(find custom_arm)/config/joint_trajectory_controller.yaml"/>
    
    <!-- position & effort controller -->
    <!-- <node name="controller_spawner" pkg="controller_manager" type="spawner"  args="joint_1_controller 
                                                                                   joint_2_controller 
                                                                                   joint_3_controller 
                                                                                   joint_4_controller"/> -->
    <!-- joint trajectory controller -->
    <node name="controller_spawner" pkg="controller_manager" type="spawner" args="custom_arm_controller" />

</launch>